numpy-stubs/__init__.pyi

```diff
@@ -82,6 +83,19 @@ _DtypeLike = Union[
 
 _NdArraySubClass = TypeVar("_NdArraySubClass", bound=ndarray)
 
+_ArrayLike = TypeVar(
+    "_ArrayLike",
+    str,
+    int,
+    float,
+    bool,
+    object,
```

Comment:
Is there a meaningful difference between `TypeVar("_ArrayLike")`,  `TypeVar("_ArrayLike", object)` and what you wrote here?

I agree that it's a good idea to clearly identify "array like" arguments, but I'm not certain this the right annotation. Note that in most cases NumPy is fine with absolutely *any* type of object in cases where it needs an "array like" argument

Reply:
You have a good point. I was already hesitating on how to create this TypeVar, but since an `array_like` can be just about anything, it makes better sense to just declare it as `_ArrayLike = TypeVar("_ArrayLike")`

Best thing is, we can always change this TypeVar if we come to another solution.